backend/src/DataAccessLayer/dalMain.cs

```diff
@@ -35,16 +35,53 @@ public CosmosDal(Secrets secrets, Config config)
             cosmosDetails = new CosmosConfig
             {
                 CosmosCollection = secrets.CosmosCollection,
+                CosmosLease = secrets.CosmosLease,
                 CosmosDatabase = secrets.CosmosDatabase,
                 CosmosKey = secrets.CosmosKey,
                 CosmosUrl = secrets.CosmosServer,
                 Retries = config.Retries,
                 Timeout = config.Timeout,
             };
 
-            // create the CosmosDB client and container
-            cosmosDetails.Client = OpenAndTestCosmosClient(secrets.CosmosServer, secrets.CosmosKey, secrets.CosmosDatabase, secrets.CosmosCollection).GetAwaiter().GetResult();
-            cosmosDetails.Container = cosmosDetails.Client.GetContainer(secrets.CosmosDatabase, secrets.CosmosCollection);
+            // create the CosmosDB client and source container
+            cosmosDetails.Client = new CosmosClient(secrets.CosmosServer, secrets.CosmosKey, cosmosDetails.CosmosClientOptions);
+            cosmosDetails.SourceContainer = cosmosDetails.Client.GetContainer(secrets.CosmosDatabase, secrets.CosmosCollection);
+            cosmosDetails.LeaseContainer = cosmosDetails.Client.GetContainer(secrets.CosmosDatabase, secrets.CosmosLease);
+            cosmosDetails.ChangeFeedProcessor = StartChangeFeedProcessorAsync(cosmosDetails).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Start the Change Feed Processor to listen for changes and process them with the HandleChangesAsync implementation
+        /// </summary>
+        /// <param name="cosmosDetails">Cosmos details</param>
+        /// <returns>ChangeFeedProcessor</returns>
+        private static async Task<ChangeFeedProcessor> StartChangeFeedProcessorAsync(CosmosConfig cosmosDetails)
+        {
+            ChangeFeedProcessor changeFeedProcessor = cosmosDetails.SourceContainer
+                .GetChangeFeedProcessorBuilder<ClientStatus>(processorName: "clientStatusProcessor", onChangesDelegate: HandleChangesAsync)
+                .WithInstanceName("clientStatusProcessorInstance")
+                .WithLeaseContainer(cosmosDetails.LeaseContainer)
+                .Build();
+

```

Comment:
I think I know how we can handle the different object types.  We'll use a range value.  However, we'll do that under the LodeRunner updates when we merge the data access code.  For now, let's just leave this as is with the ClientStatus objects and we'll generalize the processor type, but get specific on the range in the data access we are developing in LR.